src/apps/service/apps/agave.clj

```diff
@@ -68,33 +74,60 @@
     (when-not (util/uuid? app-id)
       (.getApp agave app-id)))
 
+  (getAppJobView [_ system-id app-id]
+    (validate-system-id system-id)
+    (.getApp agave app-id))
+
   ;; FIXME: remove the third parameter when we can.
   (getAppDetails [_ app-id _]
     (when-not (util/uuid? app-id)
       (.getAppDetails agave app-id)))
 
+  ;; FIXME: remove the fourth parameter when we can.
+  (getAppDetails [_ system-id app-id _]
+    (validate-system-id system-id)
+    (.getAppDetails agave app-id))
+
   (isAppPublishable [_ app-id]
     (when-not (util/uuid? app-id)
       false))
 
+  (isAppPublishable [_ system-id app-id]
+    false)
+
   (getAppTaskListing [_ app-id]
     (when-not (util/uuid? app-id)
       (.listAppTasks agave app-id)))
 
+  (getAppTaskListing [_ system-id app-id]
+    (validate-system-id system-id)
+    (.listAppTasks agave app-id))
+
   (getAppToolListing [_ app-id]
     (when-not (util/uuid? app-id)
       (.getAppToolListing agave app-id)))
 
+  (getAppToolListing [_ system-id app-id]
+    (validate-system-id system-id)
+    (.getAppToolListing agave app-id))
+
   (getAppInputIds [_ app-id]
     (when-not (util/uuid? app-id)
       (.getAppInputIds agave app-id)))
 
+  (getAppInputIds [_ system-id app-id]
+    (validate-system-id system-id)
+    (.getAppInputIds agave app-id))
+
   (formatPipelineTasks [_ pipeline]
     (pipelines/format-pipeline-tasks agave pipeline))
 
   (listJobs [self params]
     (job-listings/list-jobs self user params))
 
+  ;; TODO: Determine how this should be refactored now that we have system IDs. If I remember correctly,
+  ;; this method is used during job listings. If that's the case, we can filter apps by execution system
+  ;; easily enough.
```

Comment:
I was going to do this under the same ticket. The comment is just a reminder for myself.